### PR TITLE
Added Filter by Diet.Restricted Ingredients.

### PR DESCRIPTION
**Pull request by mertrassa**
With this PR, my main use case should be fully completed. Now, recipes with ingredients that are in diet res list will not be shown to the user. 
However, to write covering tests/change existing tests accordingly, I need to see what edge cases can happen, which I can't right now because of unfinished use cases / other waiting PRs.

### CODE REVIEW

File: src/main/java/logic/generate_recipe/generate_with_inventory/GenerateWithInventoryInteractor.java [hunk 1]
```diff
@@ -1,29 +1,48 @@
...
+import logic.dietary_restriction.DietaryRestrictionChecker;
+import logic.dietary_restriction.diet_res_ingredients.DietResDataAccessInterface;
+
 public class GenerateWithInventoryInteractor implements GenerateWithInventoryInputBoundary {
 
     private final InventoryReader inventoryReader;
     private final RecipeGateway recipeGateway;
     private final GenerateWithInventoryOutputBoundary presenter;
+    private final DietResDataAccessInterface dietResDataAccessInterface;
+    private final DietaryRestrictionChecker dietaryRestrictionChecker;
```
Comment (MZH006) [inline src/main/java/logic/generate_recipe/generate_with_inventory/GenerateWithInventoryInteractor.java]: This violates Dependency Inversion Principle.
You should create an interface:
public interface DietaryRestrictionCheckerInterface {
    boolean containsRestrictedIngredient(Recipe recipe, List<Ingredient> restrictions);
}
Then have DietaryRestrictionChecker implements DietaryRestrictionCheckerInterface

Please confirm if I am right.

File: src/main/java/logic/dietary_restriction/DietaryRestrictionChecker.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    /**
+     * Checks if a recipe contains any ingredient from the list of restrictions.
+     * Uses case-insensitive comparison on ingredient names.
+     *
+     * @param recipe       The recipe to check.
+     * @param restrictions The list of restricted ingredients.
+     * @return true if the recipe contains a restricted ingredient, false otherwise.
+     */
+    public boolean containsRestrictedIngredient(Recipe recipe, List<Ingredient> restrictions) {
+        boolean foundRestriction = false;
```
[the remark this answers is not in the document]
Comment (MZH006) [reply src/main/java/logic/dietary_restriction/DietaryRestrictionChecker.java]: Ohhhhh okay then you can keep it

**Comment by UniversalSaver** [comment]
In theory this doesn't break SRP, as the interactor would change for a single reason - A change in how recipes are mainly searched for. This type of change would be requested by a user, and only for changes in ways to search.

In practice, I think it might be preferable to use wrappers as a design pattern, slowly chipping away at the total recipe count with each filter, but that's more complicated and can be done in a refactor.

**Comment by MZH006** [comment]
Okay thank you guys for your clarification.
Resolve the conflicts and Ill approve.

**Comment by mertrassa** [comment]
Okay, I resolved the conflicts

**Review by MZH006 — APPROVED** [review]
